src/backend/gpopt/translate/CTranslatorScalarToDXL.cpp

```diff
@@ -2433,8 +2433,10 @@ CTranslatorScalarToDXL::CreateIDatumFromGpdbDatum(CMemoryPool *mp,
 	ULONG length = md_type->Length();
 	if (!md_type->IsPassedByValue() && !is_null)
 	{
-		INT len =
-			dynamic_cast<const CMDTypeGenericGPDB *>(md_type)->GetGPDBLength();
+		const CMDTypeGenericGPDB *mdtype =
```

Comment:
Maybe change for clarity?
```suggestion
		const CMDTypeGenericGPDB *md_gpdb_type =
```

Reply:
renamed